src/data/prod.lean

```diff
@@ -39,11 +39,20 @@ funext $ map_snd f g
 Composing a `prod.map` with another `prod.map` is equal to
 a single `prod.map` of composed functions.
 -/
-lemma comp_map {ε ζ : Type*}
+lemma map_comp_map {ε ζ : Type*}
   (f : α → β) (f' : γ → δ) (g : β → ε) (g' : δ → ζ) :
   prod.map g g' ∘ prod.map f f' = prod.map (g ∘ f) (g' ∘ f') :=
 rfl
 
+/--
+Composing a `prod.map` with another `prod.map` is equal to
+a single `prod.map` of composed functions, fully applied.
+-/
+lemma map_map {ε ζ : Type*}
```

Comment:
```suggestion
@[simp] lemma map_map {ε ζ : Type*}
```
Since `list.map_map` and `multiset.map_map` are both `simp`

Reply:
OK I guess so since this is more of a core type, even though we have other `map_map` lemmas in mathlib that are not simp, like in rings etc. In general, I have a larger question of why these lemmas all don't just come out of functor instances, but that's a separate discussion.

Reply:
Linter complained that this is already simp-able.